KubeArmor/core/kubeUpdate.go

```diff
@@ -2382,12 +2393,103 @@ func (dm *KubeArmorDaemon) removeBackUpPolicy(name string) {
 // == Default Posture == //
 // ===================== //
 
-func validateDefaultPosture(key string, ns *corev1.Namespace, defaultPosture string) string {
+func (dm *KubeArmorDaemon) updatEndpointsWithCM(cm *corev1.ConfigMap, action string) {
+	dm.EndPointsLock.Lock()
+	defer dm.EndPointsLock.Unlock()
+
+	dm.DefaultPosturesLock.Lock()
+	defer dm.DefaultPosturesLock.Unlock()
+
+	// get all namespaces
+	nsList, err := K8s.K8sClient.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		kg.Err("unable to fetch requested namespace")
+	}
+
+	// for each namespace if needed change endpoint depfault posture
+	for _, ns := range nsList.Items {
+		ns := ns
```

Comment:
Why this line is needed ?

Reply:
this is to avoid security concerns as `ns` is loop variable and we are passing it as a reference in helping functions, that is why it's been copied to local variable and that is being used in subsequent functions.